tests/test_container_provider.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+from mock import Mock
+
+from ansible.module_utils.basic import AnsibleModule
+
+from miqclient.api import API
+import manageiq
+
+
+@pytest.fixture(autouse=True)
+def miq_api_class(monkeypatch):
+    miq_api_class = Mock(spec=API)
+    monkeypatch.setattr("manageiq.MiqApi", miq_api_class)
+    yield miq_api_class
+
+
+@pytest.fixture
+def the_provider():
+    the_provider = Mock()
+    the_provider.name = "The provider"
+    yield the_provider
+
+
+@pytest.fixture
+def miq_ansible_module():
+    miq_ansible_module = Mock(spec=AnsibleModule)
+    yield miq_ansible_module
+
+
+@pytest.fixture
+def endpoints(miq):
+    endpoints = miq.generate_endpoints("The hostname", "12345", "The token",
+                                       "The HHostname", "54321")
+    yield endpoints
+
+
+class AnsibleModuleFailed(Exception):
+    pass
+
+
+@pytest.fixture()
+def miq(miq_api_class, miq_ansible_module, the_provider):
+    miq_ansible_module.params = {'metrics': True}
+
+    def fail(msg):
+        raise AnsibleModuleFailed(msg)
+
+    miq_ansible_module.fail_json = fail
+    miq = manageiq.ManageIQ(miq_ansible_module, "http://themanageiq.tld",
+                            "The username", "The password")
+
+    miq_api_class.return_value.post.return_value = dict(results=[dict(id=0)])
```

Comment:
a. do we want to return more than the id of the added provider in the tests? not required at this step anyway. this is a reponse example:
```
[{'api_version': u'v1',
               'created_on': u'2016-09-08T15:11:29Z',
               'guid': u'84a4084a-75d6-11e6-92bc-0242ee817803',
               'id': 134,
               'name': u'Openshift01',
               'tenant_id': 1,
               'type': u'ManageIQ::Providers::Openshift::ContainerManager',
               'updated_on': u'2016-09-08T15:11:29Z',
               'zone_id': 1}]}
```

b. the update provider operation is also done with post request, so when we'll add tests for updating a provider maybe we'll need a way to differentiate between the two operations.also, maybe we could look at the `'updated_on'` key